src/gateway/bridge/mod.rs

```diff
@@ -97,9 +98,16 @@ pub struct ShardRunnerInfo {
 /// Newtype around a callback that will be called on every incoming request. As long as this
 /// collector should still receive events, it should return `true`. Once it returns `false`, it is
 /// removed.
-pub struct CollectorCallback(pub Box<dyn Fn(&Event) -> bool + Send + Sync>);
+#[derive(Clone)]
+pub struct CollectorCallback(pub Arc<Box<dyn Fn(&Event) -> bool + Send + Sync>>);
 impl std::fmt::Debug for CollectorCallback {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_tuple("CollectorCallback").finish()
     }
 }
+
+impl PartialEq for CollectorCallback {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::as_ptr(&self.0) == Arc::as_ptr(&other.0)
```

Comment:
Arc has a method for this, `Arc::ptr_eq`.

Reply:
It has, but the conversion does not work as expected. This method failed my tests.

Reply:
I very much doubt that... https://doc.rust-lang.org/stable/src/alloc/sync.rs.html#1786

Reply:
I did not realise, this fas fixed, because it was a problem some time ago: https://github.com/rust-lang/rust/issues/103763
I will change it.